### ISSUE TITLE
missing module

### ISSUE THREAD

**Issue by dhdurgee** [benallard/galileo#2]
I had been using galileo on another system that is currently down.  Given this I attempted to install on another system.  On the other system the command I was using is:

~/Src/benallard-galileo-83d19bece241/run -v --force

On the new system the directory name is different, but the attempt to run yields:

dhdurgee@DG41TY ~/Downloads $ ~/src/galileo-master/run -v --force
Traceback (most recent call last):
  File "/home/dhdurgee/src/galileo-master/run", line 3, in <module>
    from galileo.main import main
  File "/perm/dhdurgee/src/galileo-master/galileo/main.py", line 13, in <module>
    import requests
ImportError: No module named requests
dhdurgee@DG41TY ~/Downloads $ 

I have been using your program for quite a while on the other system without problem.  Why am I encountering this problem?  How can it be corrected?

Dave

**Comment by benallard** [comment]
you had the `python-request` package installed on the other system ?

**Comment by dhdurgee** [comment]
You are correct, I missed installing your prerequsites.  It has been long enough since I installed your package I forgot what was involved.

I now have managed to shoot myself in the foot and perhaps you can tell me how to correct it.  I decided it might be simpler to use pip to install your package and requirements.  I checked and found linux mint had python-pip in their repository, so I installed it.  They do not have pyusb or requests, so I was going to have to use pip to retrieve them.  I started with a pip list, which told me that since mint installed 8.1.1 that I should use pip to upgrade itself via "pip install --upgrade pip" to do so.  I ran this via sudo and got a message:  "The directory '/home/dhdurgee/.cache/pip/http' or its parent directory is not owned by the current user and the cache has been disabled. Please check the permissions and owner of that directory. If executing pip with sudo, you may want sudo's -H flag.
The directory '/home/dhdurgee/.cache/pip' or its parent directory is not owned by the current user and caching wheels has been disabled. check the permissions and owner of that directory. If executing pip with sudo, you may want sudo's -H flag." and it then went on to install the new pip.  Unfortunately since then I find pip unusable.  Attempting to run it yields "dhdurgee@DG41TY ~/Downloads $ pip list
Traceback (most recent call last):
  File "/usr/bin/pip", line 11, in <module>
    sys.exit(main())
  File "/usr/local/lib/python2.7/dist-packages/pip/__init__.py", line 16, in main
    from pip._internal.utils.entrypoints import _wrapper
  File "/usr/local/lib/python2.7/dist-packages/pip/_internal/utils/entrypoints.py", line 3, in <module>
    from pip._internal.cli.main import main
  File "/usr/local/lib/python2.7/dist-packages/pip/_internal/cli/main.py", line 60
    sys.stderr.write(f"ERROR: {exc}")
                                   ^
SyntaxError: invalid syntax
dhdurgee@DG41TY ~/Downloads $"

I also just noticed in the install output the following: "Installing collected packages: pip
  Found existing installation: pip 8.1.1
    Not uninstalling pip at /usr/lib/python2.7/dist-packages, outside environment /usr
  Running setup.py install for pip ... done
Successfully installed pip-21.0.1
dhdurgee@DG41TY ~/Downloads $ "

Looking around I see pip related files in /usr/local/bin and I have a ~.cache/pip directory with files in it.

How do I undo this mess?  How should I have updated pip to get the new version installed in /usr/bin?

**Comment by benallard** [comment]
I don't believe pip can install itself in /usr/bin. that directory is reserved for official packages of your distribution you shouldn't have needed to update pip. as a thumb of rule, what you install via packages remains as package and shouldn't be updated / installed otherwise. an older version of pip usually is okay.

**Comment by dhdurgee** [comment]
Unfortunately I did attempt to update pip and have now damaged my ability to use pip at all.  How can I undo this mess?  Can I simply remove that ~/.cache/pip directory and the /usr/local/bin entries, or are there some other environment or configuration files that need to be corrected or removed?

**Comment by dhdurgee** [comment]
I have worked around this problem by determining how to install the dependencies by hand.  I was just able to sync fitbits with your package.

I should note error that I had to correct.  The "99-fitbit.rules" as provided cannot just be copied to the udev rules directory.  I found I had to edit it to match the line issued by the package as it ran, which is 'SUBSYSTEM=="usb", ATTR{idVendor}=="2687", ATTR{idProduct}=="fb01", SYMLINK+="fitbit", MODE="0666"' instead.
